src/engine/Match_search_mode.ml

```diff
@@ -523,17 +523,18 @@ let children_explanations_of_xpat (env : env) (xpat : Xpattern.t) : ME.t list =
 
 let rec filter_ranges (env : env) (xs : RM.ranges) (cond : R.metavar_cond) :
```

Comment:
Do we want to allow this?
```yaml
rules:
- id: test
  severity: INFO
  message: Test
  languages: [python]
  patterns:
  - pattern: $A + $B
  - metavariable-pattern:
      metavariable: $B
      patterns:
      - pattern: bar($X)
  - metavariable-pattern:
      metavariable: $X
      pattern: x
```
because this allows that, so the other in which `metavariable-pattern`s are specified will alter the meaning of the rule.
cc @aryx